join-statements/joinstatements.md

```diff
@@ -0,0 +1,423 @@
+
+2) Assuming 2 tables, A and B:
+
+**CROSS JOIN** - it shows all possible combinations between rows from table A and rows from table B, so a result from the query would be <# of A rows> x <# of B rows>
+
+A real world example would be if I had a table with fruit juices, and another table with entrees at a high school cafeteria. A CROSS JOIN would allow me to compare all the possible combinations of meals for the high schoolers.
+
+```
+SELECT m.dish, d.drink
+FROM meals AS m
+CROSS JOIN drink AS d;
+```
+
+**INNER JOIN** - it's the result of a query between 2 tables where the output is the intersection of table A and table B that satisfies a specified conditional statement between both tables using the `ON` statement.
+
+A real world example would be if, there were a table of students being specified in the `FROM` portion of the query, and another table of different extracurricular societies that keeps records of students by their student ID. We could use the `INNER JOIN`, to get a list of students that are found in a given society using an INNER JOIN between these two tables. The `INNER JOIN` would return matches as specified in the `ON` statement. The query would look something like this.
+
+```
+SELECT students.name societies.society
+FROM students
+INNER JOIN societies ON students.id = societies.student_id;
+```
+
+**LEFT OUTER JOIN**- similar to the INNER JOIN, but assuming table A is specified in the `FROM` statement, in the output query, there would be a row for every row in table A. If the conditional statement specified in `ON` is not met, then this row output null values in the columns of table B.
+
+Similar to the setup of the INNER JOIN, except if we do a LEFT OUTER JOIN instead, we would get the students already in the extracurricular society and also any student that is not in the extracurricular activity. This could be good in order to see which students could be good recruiting targets.
+
+```
+SELECT students.name societies.society
+FROM students
+LEFT JOIN societies ON students.id = societies.student_id;
+```
+
+**RIGHT OUTER JOIN** - similar to LEFT OUTER JOIN, except instead of focusing on the table specified in the `FROM` statement (in the previous case it was table A), we would focus on the table linked to the  `RIGHT OUTER JOIN` statement, in this case, table B. Similarly, there will be a single row in the output for every row in table B, and any row that does not meet the conditional in the `ON` statement will have null values in the output values of columns from table A.
+
+In the extracurricular society example shown before, using a RIGHT OUTER JOIN would yield an output similar to the INNER JOIN output, _plus_ if we've got societies with no members, they would show up. This would be useful for example to pinpoint societies for students interested in building a society from the ground up.
```

Comment:
not the same, does not meet desired goal

Reply:
done